x-pack/plugins/canvas/server/usage/collector_helpers.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/*
+ * @param ast: an ast that includes functions to track
+ * @param cb: callback to do something with a function that has been found
+ */
+
+export interface AST {
```

Comment:
Note that the AST can also be an empty object, that's what it is with an empty expression. Not really knowing TS well, I'm not sure if that affects this interface or not, so it seemed worth mentioning.

Reply:
@w33ble  `fromExpression('')` gives a response of `{type: 'expression', chain: []}`

Is that what you meant by an empty expression? Or when can it just be an empty object?

Reply:
Ah, I guess I was mistaken, I didn't think it contained any keys at all. Nevermind then, this interface is correct.